lib/pxlogger.js

```diff
@@ -4,27 +4,25 @@
  */
 
 class PxLogger {
-    constructor() {
-        this.debugMode = false;
-        this.appId = '';
-    }
+  constructor(params) {
+    this.debugMode = params.debugMode || false;
+    this.appId = params.pxAppId || "PX_APP_ID";
+    this.internalLogger = params.customLogger || console;
+  }
 
-    init(pxConfig) {
-        this.debugMode = pxConfig.conf.DEBUG_MODE;
-        this.appId = pxConfig.conf.PX_APP_ID;
+  debug(msg) {
+    if (this.debugMode && msg) {
```

Comment:
What about validating that msg is a string, the same as in error.

Reply:
yup, I can make that change 👍